### PR TITLE
Made tweaks to the templating system to allow some more flexibility with channels.

### PR DESCRIPTION
**Pull request by pelson**
Note: Breaks use of channels/target in favour of the more flexible channels/targets.

### CODE REVIEW

**Comment by pelson** [comment]
I've been using this on the obvious-ci feedstock.

File: conda_smithy/feedstock_content/ci_support/upload_or_check_non_existence.py [hunk 1]
```diff
@@ -44,7 +44,7 @@ def upload(cli, meta, owner, channels):
     try:
         with open('binstar.token', 'w') as fh:
             fh.write(cli.token)
-        subprocess.check_call(['binstar', '--quiet', '-t', 'binstar.token',
+        subprocess.check_call(['anaconda-client', '--quiet', '-t', 'binstar.token',
```
Comment (ocefpaf) [inline conda_smithy/feedstock_content/ci_support/upload_or_check_non_existence.py]: Should this be only anaconda as the change in https://github.com/conda-forge/obvious-ci-feedstock/pull/3/files#r43991411?

Comment (pelson) [inline conda_smithy/feedstock_content/ci_support/upload_or_check_non_existence.py]: Yep. My bad.

Comment (pelson) [inline conda_smithy/feedstock_content/ci_support/upload_or_check_non_existence.py]: I'm saving up a few changes and will submit a PR once the Obvious-ci feedstock is doing its business.

**Pull request merged**